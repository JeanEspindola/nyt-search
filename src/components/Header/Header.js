/*
* Header Presentation Component
*
* Displays the header top bar.
*
* */
import React from 'react';
import PropTypes from 'prop-types';
import { Row } from 'react-bootstrap';
import WithLocale from '../HOC/WithLocale';
import './Header.css';

const Header = ({ locale }) => (
  <Row className="Header">
    <div className="Header__Title">
      {locale.header_title}
    </div>
  </Row>
);

Header.propTypes = {
  locale: PropTypes.object,
};

export default WithLocale(Header);
