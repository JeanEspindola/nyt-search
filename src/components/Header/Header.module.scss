.header {
  display: flex;
  border-top: 1px solid #ebebeb;
  margin-top: 5px;
  border-bottom: 4px double #000;

  .title {
    font-size: 20px;
    margin: auto;
  }

}

.logo {
  position: absolute;
  top: 8px;
}
