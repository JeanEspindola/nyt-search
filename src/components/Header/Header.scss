.Header {
  text-align: center;
  background-color: #222;
  line-height: 35px;

  .Header__Title {
    color: white;
    font-size: 16px;
    top: 50%;
  }
}