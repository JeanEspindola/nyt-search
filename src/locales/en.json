{
  "go_to_results": "Go to Results Page",
  "header_title": "\"The New York Times\" article search application",
  "next_page": "Next Page",
  "no_details": "No details to show.",
  "no_results": "There are no results to display.",
  "previous_page": "Previous Page",
  "read_article": "Read the Full Article",
  "results": "Results:",
  "search_terms": "Search your terms here.",
  "type_query": "Type search query term in here:"
}