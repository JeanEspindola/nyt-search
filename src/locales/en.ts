// eslint-disable-next-line import/no-anonymous-default-export
export default {
  goToResults: "Go to Results Page",
  headerTitle: "\"The New York Times\" article search application",
  nextPage: "Next Page",
  noDetails: "No details to show.",
  noResults: "There are no results to display.",
  previousPage: "Previous Page",
  readArticle: "Read the Full Article",
  results: "Results:",
  searchTerms: "Search your terms here.",
  typeQuery: "Type search query term in here:"
}
