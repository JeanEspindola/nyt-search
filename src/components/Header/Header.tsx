/*
* Header Presentation Component
*
* Displays the header top bar.
*
* */
import { Row } from 'react-bootstrap';
import { FormattedMessage } from 'react-intl'
import './Header.css';

const Header = () => (
  <Row className="Header">
    <div className="Header__Title">
      <FormattedMessage id="headerTitle"/>
    </div>
  </Row>
);

export default Header;
