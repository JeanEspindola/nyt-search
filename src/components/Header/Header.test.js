import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Header from './Header';
import en from '../../locales/en.json';

describe('Header', () => {
  let Component;

  beforeEach(() => {
    Component = shallow(
      <Header />,
      { context: { locale: en } },
    )
      .find('Header')
      .shallow();
  });

  it('renders Header without crashing', () => {
    expect(Component.length).toBeTruthy();
    expect(shallowToJson(Component)).toMatchSnapshot();
  });
});
