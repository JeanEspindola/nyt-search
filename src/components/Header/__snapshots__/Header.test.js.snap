// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header without crashing 1`] = `<Header />`;
